WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift

```diff
@@ -27,6 +27,7 @@ final class OrderDetailsViewModel {
     func update(order newOrder: Order) {
         self.order = newOrder
         dataSource.update(order: order)
+        editNoteViewModel.update(order: order)
```

Comment:
As we are using a unique view model, we have to update when the stored order is updated to be synchronised.